build.gradle

```diff
@@ -1,46 +1,123 @@
+import org.grails.gradle.plugin.doc.PublishGuideTask
+
+buildscript {
+    repositories {
+        maven { url "https://repo.grails.org/grails/core" }
+        mavenCentral()
+    }
+    dependencies {
+        classpath "org.grails.grails-doc:org.grails.grails-doc.gradle.plugin:$grailsVersion"
+        classpath "org.grails:grails-gradle-plugin:$grailsVersion"
+    }
+}
+
 plugins {
-    id 'groovy'
-    id 'java-library'
-    id 'io.github.gradle-nexus.publish-plugin'
-    id 'maven-publish'
-    id 'signing'
+    id "java-library"
+    id "eclipse"
+    id "idea"
 }
 
+apply plugin: "org.grails.grails-plugin"
+apply plugin: 'org.grails.grails-doc'
+apply plugin: 'org.grails.grails-publish'
+
+
 version =  projectVersion
 group = 'org.grails.plugins'
-ext.set('grailsVersion', libs.versions.grails.get())
-ext.set('isReleaseVersion', !version.toString().endsWith('-SNAPSHOT'))
-ext.set('isSnapshot', !isReleaseVersion)
-
-apply plugin: 'org.grails.grails-plugin' // Needs to be applied after grailsVersion has been set
 
 repositories {
   mavenLocal()
   mavenCentral()
   maven { url = 'https://repo.grails.org/grails/core' }
 }
 
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(17)
+    }
+}
+
+dependencyManagement {
+    imports {
+        mavenBom "org.grails:grails-bom:$grailsVersion"
+    }
+    applyMavenExclusions false
+}
+repositories {
+    mavenCentral()
+    maven { url "https://repo.grails.org/grails/core" }
+}
+
 dependencies {
 
-    api libs.quartz
+    api "org.quartz-scheduler:quartz:$quartzVersion"
 
-    implementation libs.grails.bootstrap
-    implementation libs.grails.core
-    implementation libs.groovy.sql
-    implementation libs.spring.beans
-    implementation libs.spring.context
-    implementation libs.spring.context.support
+    implementation "org.grails:grails-bootstrap"
+    implementation "org.grails:grails-core"
+    implementation "org.apache.groovy:groovy-sql"
+    implementation "org.springframework:spring-beans"
+    implementation "org.springframework:spring-context"
+    implementation "org.springframework:spring-context-support"
+    compileOnly "org.apache.groovy:groovy"
+    compileOnly "jakarta.annotation:jakarta.annotation-api"
 
-    compileOnly libs.groovy.core // CompileStatic and Slf4j
-    compileOnly libs.javax.annotation.api // Provided
+    testImplementation "org.spockframework:spock-core"
 
-    testImplementation libs.spock.core
+    testRuntimeOnly "org.slf4j:slf4j-nop:$slf4jVersion" // Get rid of warning about missing slf4j implementation during test task
+}
+
+tasks.named('bootJar') {
+    enabled = false // Plugins should not create a bootJar
+}
+tasks.named('jar', Jar) {
+    enabled = true // Enable the jar task again, as the bootJar task has been disabled
+    archiveClassifier = '' // Remove '-plain' suffix from jar file name
```

Comment:
By default `-plain` should be removed by the grails plugin.

Reply:
deleted